backslash/contrib/slash_plugin.py

```diff
@@ -145,9 +145,35 @@ def session_interrupt(self):
         if self.session is not None:
             self.session.report_interrupted()
 
+    @slash.plugins.register_if(_HAS_REPORT_UPCOMING_TESTS)
+    @handle_exceptions
+    def report_upcoming_tests(self):
```

Comment:
@maormarcus I wouldn't call this hook "report_upcoming_tests". This is the action to be performed, not the name of this point in time. A better name would probably be "tests_collected" perhaps?